src/Data/Array/Mutable/Unlifted/Linear.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UnboxedTuples #-}
+{-# LANGUAGE UnliftedNewtypes #-}
+
+-- |
+-- This module defines a linear wrapper around 'GHC.MutableArray#'
+-- which can be manipulated in pure contexts.
+--
+-- It ensures that effects happen in order. However, access to
```

Comment:
> It ensures that effects happen in order

Of course it does: officially, there is no effect. This is more a note to your past self than documentation.

Reply:
You are right, I will remove that sentence.